metrics-operator/controllers/common/providers/prometheus/prometheus.go

```diff
@@ -3,35 +3,37 @@ package prometheus
 import (
 	"context"
 	"encoding/json"
-	"fmt"      //nolint:gci
-	"net/http" //nolint:gci
+	"fmt"
+	"net/http"
 	"time"
 
 	"github.com/go-logr/logr"
 	metricsapi "github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha3"
 	promapi "github.com/prometheus/client_golang/api"
 	prometheus "github.com/prometheus/client_golang/api/prometheus/v1"
 	"github.com/prometheus/common/model"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var errCouldNotCast = fmt.Errorf("could not cast result")
 var errNoValues = fmt.Errorf("no values in query result")
 var errTooManyValues = fmt.Errorf("too many values in query result")
 
+type RountripGetter func(context.Context, metricsapi.KeptnMetricsProvider, client.Client) (http.RoundTripper, error)
+
 type KeptnPrometheusProvider struct {
-	Log        logr.Logger
-	HttpClient http.Client
+	Log       logr.Logger
+	K8sClient client.Client
```

Comment:
Let's add the function here

```suggestion
	K8sClient client.Client
	getter RountripGetter
```